backend/deepchecks_monitoring/resources.py

```diff
@@ -129,7 +130,7 @@ def redis_settings(self) -> config.RedisSettings:
     @property
     def settings(self) -> config.Settings:
         """Get settings."""
-        if not isinstance(self._settings, config.Settings):
+        if not isinstance(self._settings, BaseSettings):
```

Comment:
hm, why?

Reply:
Because WorkerSettings does not inherit config.Settings, 
take a look at bgtasks/actors.py, WorkerBootstrap
it uses WorkerSettings which is defined above.

Reply:
I see, but I think the `Worker` should not access `ResourcesProvider.settings` directly, instead it should use properties like `ResourcesProvider.email_settings`, `ResourcesProvider.database_settings`

if I remember correctly the idea was to make it possible to pass only a subset of settings to the resources provider to be able to use only a subset of its functionality, so, for example, if you try to access `ResourcesProvider.email_settings` on a resources provider instance that was not initialized with email settings, you will get an exception

Reply:
Ok, im not sure that i understand what youre asking because execute_monitor does access settings directly, and also the notificator accesses settings.deployment_url which does not exist in WorkerSettings.
i can change them not to do the above, is that what you meant?